### PR TITLE
Add a table to show how changes in membership should be interpreted

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/matrix-org/matrix-doc/issues/876

### CODE REVIEW

File: event-schemas/schema/m.room.member [hunk 1]
```diff
@@ -21,6 +21,29 @@ description: |-
...
+  .. TODO: Improve how this table is written? We use a csv-table to get around vertical header restrictions.
+
+  .. csv-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    "","to ``invite``","to ``join``","to ``leave``","to ``ban``","to ``knock``"
+    "from ``invite``","No change.","User joined the room.","If the ``state_key`` is the same as the ``sender``, the user rejected the invite. Otherwise, the ``state_key`` user had their invite revoked.","User was banned.","Not implemented."
+    "from ``join``","Must never happen.","``displayname`` or ``avatar_url`` changed.","If the ``state_key`` is the same as the ``sender``, the user left. Otherwise, the ``state_key`` user was kicked.","User was kicked and banned.","Not implemented."
+    "from ``leave``","New invitation sent.","User joined.","Must never happen.","User was banned.","Not implemented."
```
Comment (uhoreg) [inline event-schemas/schema/m.room.member]: `leave` -> `leave` can happen if, e.g. two mods (from different homeservers) kick a user at the same time, or a mod kicks a user at the same time that that the user leaves, in which case the client would see two `leave` events.

Comment (turt2live) [reply event-schemas/schema/m.room.member]: hmm, good point. Should we just label it as "No change"?

File: event-schemas/schema/m.room.member [hunk 2]
```diff
@@ -21,6 +21,29 @@ description: |-
...
+
+  .. TODO: Improve how this table is written? We use a csv-table to get around vertical header restrictions.
+
+  .. csv-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    "","to ``invite``","to ``join``","to ``leave``","to ``ban``","to ``knock``"
+    "from ``invite``","No change.","User joined the room.","If the ``state_key`` is the same as the ``sender``, the user rejected the invite. Otherwise, the ``state_key`` user had their invite revoked.","User was banned.","Not implemented."
+    "from ``join``","Must never happen.","``displayname`` or ``avatar_url`` changed.","If the ``state_key`` is the same as the ``sender``, the user left. Otherwise, the ``state_key`` user was kicked.","User was kicked and banned.","Not implemented."
```
Comment (KitsuneRal) [inline event-schemas/schema/m.room.member]: Can we make two separate columns for `state_key == sender` and `state_key != sender` for transitions to `leave`? Should be a bit more structured and easy to consume.
Also: probably just drop `knock` from the table for now?

Comment (turt2live) [reply event-schemas/schema/m.room.member]: I'd rather not split it into more columns as the rendering is already pretty messy.

I also think that knock should be here to reinforce that it doesn't do anything.

**Pull request merged**